src/slicer.cpp

```diff
@@ -671,20 +710,53 @@ Slicer::Slicer(Mesh* mesh, int initial, int thickness, int layer_count, bool kee
             if (layer_nr < 0) continue;
             
             SlicerSegment s;
+            int end_edge_start_pt = -1;
             if (p0.z < z && p1.z >= z && p2.z >= z)
+            {
                 s = project2D(p0, p2, p1, z);
+                end_edge_start_pt = 0;
+                if (p1.z == z)
+                {
+                    s.endOtherFaces = v1.connected_faces;
```

Comment:
I'll have to look at it more, but my initial reaction is yes.  I'll look into changing it.